app/data_model/answer_store.py

```diff
@@ -4,7 +4,12 @@
 
 
 class Answer(object):
-    def __init__(self, group_id, block_id, answer_id, value, group_instance=0, answer_instance=0):
+    def __init__(self, group_id=None, block_id=None, answer_id=None, value=None, group_instance=0, answer_instance=0):
+        valid = (group_id or answer_id or block_id or value) is not None
```

Comment:
Well, from the error message raised, this was the intended behaviour - but really it only makes sense that group_id and block_id are optional. I think this may have originally been to support filtering on the answer store.